app/assets/javascripts/discourse/app/components/post-stream.gjs

```diff
@@ -0,0 +1,777 @@
+import { cached, tracked } from "@glimmer/tracking";
+import { fn, get, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { schedule } from "@ember/runloop";
+import { service } from "@ember/service";
+import { TrackedSet } from "@ember-compat/tracked-built-ins";
+import { and, not } from "truth-helpers";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
+import LoadMore from "discourse/components/load-more";
+import PostFilteredNotice from "discourse/components/post/filtered-notice";
+import discourseDebounce from "discourse/lib/debounce";
+import { bind } from "discourse/lib/decorators";
+import offsetCalculator from "discourse/lib/offset-calculator";
+import { Placeholder } from "discourse/lib/posts-with-placeholders";
+import DiscourseURL from "discourse/lib/url";
+import Post from "./post";
+import PostGap from "./post/gap";
+import PostPlaceholder from "./post/placeholder";
+import PostSmallAction from "./post/small-action";
+import PostTimeGap from "./post/time-gap";
+import PostVisitedLine from "./post/visited-line";
+
+const CLOAKABLE_CLASS = "post-stream__cloakable-item";
+const CLOAKABLE_CLASS_SELECTOR = `.${CLOAKABLE_CLASS}`;
+const DAY_MS = 1000 * 60 * 60 * 24;
+const POST_MODEL = Symbol("POST");
+const RESIZE_DEBOUNCE_MS = 100;
+const SCROLL_BATCH_INTERVAL_MS = 10;
+const SLACK_FACTOR = 5;
+
+// change this value to true to debug the eyeline position
+const DEBUG_EYELINE = true;
+
+let cloakingEnabled = true;
+const cloakingPrevented = new TrackedSet();
+
+export function disableCloaking() {
+  cloakingEnabled = false;
+}
+
+export function preventCloaking(postId) {
+  cloakingPrevented.add(postId);
+}
+
+export default class PostStream extends Component {
+  @service appEvents;
+  @service capabilities;
+  @service header;
+  @service screenTrack;
+  @service search;
+  @service site;
+  @service siteSettings;
+
+  @tracked cloakAbove;
+  @tracked cloakBelow;
+  @tracked cloakOffset;
+
+  observedPostNodes = new Set();
+  uncloakedPostNumbers = new Set();
+
+  #bottomBoundaryElement;
+  #bottomEyelineDebugElement;
+  #cloakedPostsHeight = {};
+  #cloakingObserver;
+  #currentPostElement;
+  #postsOnScreen = {};
+  #observedCloakBoundaries = {
+    above: null,
+    below: null,
+  };
+  #onScreenBoundaries = {
+    min: null,
+    max: null,
+  };
+  #viewportObserver;
+  #wrapperElement;
+
+  constructor() {
+    super(...arguments);
+
+    // initialize the cloaking offset area
+    this.#updateCloakOffset();
+
+    // TODO (glimmer-post-stream) do we need this?
+    this.appEvents.on("post-stream:posted", this, this.#postedTriggered);
+    this.appEvents.on("post-stream:refresh", this, this.#scrollTriggered);
```

Comment:
Can we try removing these and see if tests pass?

Reply:
They did.